src/lib/cht-api.ts

```diff
@@ -1,8 +1,12 @@
 import _ from 'lodash';
 import axios, { AxiosHeaders } from 'axios';
+import axiosRetry from 'axios-retry';
+import { axiosRetryConfig } from './retry-logic';
 import { UserPayload } from '../services/user-payload';
 import { AuthenticationInfo, Config, ContactType } from '../config';
 
+axiosRetry(axios, axiosRetryConfig);
```

Comment:
Does this apply globally to the default axios instance? I think maybe we should now consider creating/configuring an axios instance for reuse within `ChtApi`

Reply:
This adjusts the `axios` object within the `cht-api.ts` module. I don't think it is necessary to manage a singleton beyond what we are doing here. Is there a scenario you're concerned about?

Reply:
We could tie it directly to the session in `ChtApi` setting the appropriate base url, request timeouts etc so it's cleaner to read/write. We can also eliminate much of the request config eg `this.authorizationOptions()` that we're currently doing

Reply:
Sorry, but I don't totally understand what you're imagining here. Can you clarify both the problem you're trying to solve and the approach you're envisioning?  Sounds like you are imagining a wrapper around axios?

Reply:
Not a wrapper, just an instance of axios https://axios-http.com/docs/instance 
```
const instance = axios.create({
  baseURL: 'https://some-domain.com/api/',
  timeout: 1000,
  headers: {'X-Custom-Header': 'foobar'}
});
```

Reply:
yes. i love this suggestion. can i do after https://github.com/medic/cht-user-management/pull/75 which has a refactor already to cht-api.ts ...  i think it'll just be an easier merge